packages/edit-site/src/components/sidebar-edit-mode/template-panel/replace-template-button.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+import { __experimentalBlockPatternsList as BlockPatternsList } from '@wordpress/block-editor';
+import { MenuItem, Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEntityRecord } from '@wordpress/core-data';
+import { useAsyncList } from '@wordpress/compose';
+import { serialize } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../../store';
+
+export default function ReplaceTemplateButton( {
+	onClick,
+	availableTemplates,
+} ) {
+	const [ showModal, setShowModal ] = useState( false );
+	const onClose = useCallback( () => {
+		setShowModal( false );
+	}, [] );
```

Comment:
Nit: I think there's no need to memoize this callback. It's not an effect dependency or prop for the `memo`-ed component.

Reply:
I'd echo George on this one.